app/src/battery.c

```diff
@@ -25,40 +25,242 @@ static uint8_t last_state_of_charge = 0;
 uint8_t zmk_battery_state_of_charge() { return last_state_of_charge; }
 
 static int zmk_battery_update(const struct device *battery) {
-    struct sensor_value state_of_charge;
+    struct sensor_value state_of_charge,
+                        state_of_health,
```

Comment:
Since you're writing to bat_state_data after each channel read, you don't need a separate variable for each channel. You can just have one sensor value and keep reusing it.

Reply:
This is true. the combined bat_state_data was added to this code after I had written the rest of it.